lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -153,12 +153,20 @@ static bool isArchetypeValidInFunction(ArchetypeType *A, const SILFunction *F) {
 
 namespace {
 
-/// When resilience is bypassed, direct access is legal, but the decls are still
-/// resilient.
+/// When resilience is bypassed or package serialization is enabled, direct access
+/// is legal, but the decls are still resilient.
 template <typename DeclType>
 bool checkResilience(DeclType *D, ModuleDecl *M,
-                     ResilienceExpansion expansion) {
-  return !D->getModuleContext()->getBypassResilience() &&
+                     ResilienceExpansion expansion,
+                     bool isSerialized) {
+  // Check whether the SIL definition referencing D is serialized but
+  // in minimal resilience expansion context (e.g. M was resiliently
+  // built); this means package serialization was enabled and direct
+  // access should be allowed.
+  auto serializedInResilientMode =
+      expansion == ResilienceExpansion::Minimal && isSerialized;
```

Comment:
The resilience expansion of a ‘serialized’ function is always minimal. (See getResilienceExpansion() in SILFunction.h)

by adding this condition (‘isSerialized()‘) we will by pass the check even in regular resilient mode (no package cmo enabled). I don’t think this is desirable.

 Can you restrict this bypass to only when the current module is in the same package?

Reply:
Updated to check for that as well as whether module was built resiliently when serialized.

Reply:
Had to add SerializePackageEnabled bit to Module; the SILOpts.EnableSerializePackage info gets lost during SIL cloning stage of Package CMO, so it needed to be serialized to the binary module. It's added to this SILVerifier check function along with others we discussed.